lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -23,105 +25,155 @@ import Spinner from './Spinner';
+  const setFetched = () => {
+    count.current -= 1;
+    setFetchCount(count.current);
+  };
+
+  /**
+   * Helper to handle thrown errors from from API requests.
+   *
+   * @param {Error} e - Error object from request.
+   * @param {string} errorState - One of "error-fetch",
+   *  "error-authorizing", or "error-report-submission"
+   * @param {boolean} [retry=true] - Can the request be retried?
+   */
+  const handleError = (e, errorState, retry = true) => {
+    setFetched();
+    if (e instanceof ApiError && !e.errorMessage && retry) {
+      setErrorState('error-authorizing');
+    } else {
+      setError(e);
+      setErrorState(errorState);
+    }
+  };
+
+  /**
+   * Fetch the groups from the sync endpoint if `sync` object exists.
+   */
+  const fetchGroups = async () => {
+    if (apiSync) {
+      try {
+        setFetching();
+        const groups = await apiCall({
+          authToken,
+          path: apiSync.path,
+          data: apiSync.data,
+        });
+        rpcServer.resolveGroupFetch(groups);
+        setFetched();
+      } catch (e) {
+        handleError(e, 'error-fetch');
+      }
+    }
+  };
 
   /**
-   * Fetch the URL of the content to display in the iframe.
+   * Fetch the URL of the content to display in the iframe if `viaCallbackUrl`
+   * exists.
    *
    * This will typically be a PDF URL proxied through Via.
    */
-  const fetchContentUrl = useCallback(async () => {
+  const fetchContentUrl = async () => {
     if (!viaCallbackUrl) {
       // If no "callback" URL was supplied for the frontend to use to fetch
       // the URL, then the backend must have provided the Via URL in the
       // initial request, which we'll just use directly.
       return;
     }
-
     try {
-      setLtiLaunchState({
-        ...INITIAL_LTI_LAUNCH_STATE,
-        state: 'fetching-url',
-      });
+      setFetching();
       const { via_url: contentUrl } = await apiCall({
         authToken,
         path: viaCallbackUrl,
       });
-      setLtiLaunchState({
-        ...INITIAL_LTI_LAUNCH_STATE,
-        state: 'fetched-url',
-        contentUrl,
-      });
+      setFetched();
+      setContentUrl(contentUrl);
     } catch (e) {
-      if (e instanceof ApiError && !e.errorMessage) {
-        setLtiLaunchState({
-          ...INITIAL_LTI_LAUNCH_STATE,
-          state: 'authorizing',
-        });
-      } else {
-        setLtiLaunchState({
-          ...INITIAL_LTI_LAUNCH_STATE,
-          state: 'error',
-          error: e,
-          failedAction: 'fetch-url',
-        });
-      }
+      handleError(e, 'error-fetch');
     }
-  }, [authToken, viaCallbackUrl]);
+  };
 
   /**
-   * Fetch the assignment content URL when the app is initially displayed.
+   * Fetch the assignment content URL and groups when the app is initially displayed.
    */
   useEffect(() => {
     fetchContentUrl();
-  }, [fetchContentUrl]);
+    fetchGroups();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
In my brief manual testing today, I wasn't able to reproduce any issues when removing the two lint suppressions in this file and adding the dependencies and `useCallback(...)` hooks as directed by the lint errors.

Reply:
That may be the case, I didn't circle back around that after I got everything working, but I'll test all the error paths too. Thanks Rob.

Reply:
Update: We discussed this on Slack. The intent here was to only ever execute this on first mount. Having said that, it looks to me like you'd get that even if you did specify dependencies, since `fetchContentUrl` and `fetchGroups` never change. That might avoid confusion if for some reason in future we make a change where these callbacks' dependencies do change and we do want them to re-run.